app/src/main/java/com/wizeline/heroes/ui/fragments/CharactersFragment.kt

```diff
@@ -62,8 +70,11 @@ class CharactersFragment : Fragment() {
 
     private fun setupViewModel() {
         // Retrieving characters from retrofit
-        _marvelViewModel = ViewModelProvider(this).get(MarvelViewModel::class.java)
-        mMarvelViewModel.marvelViewState.observe(viewLifecycleOwner, {
+        val marvelViewModel: IMarvelViewModel by viewModels {
```

Comment:
Based on Android docs: https://developer.android.com/training/dependency-injection/hilt-jetpack#viewmodels

the call to `by viewModels` should be at the class level (as a class property), not inside a function. This will ensure the viewModel is properly compiled and Dependencies generated at Compile Time, instead of Runtime.

Reply:
Done.